rbx_types/src/basic_types.rs

```diff
@@ -77,27 +75,16 @@ pub struct Vector3 {
 }
 
 fn approx_unit_or_zero(value: f32) -> Option<i32> {
-    if abs_diff_eq!(value.abs(), 1.0) {
-        if value.is_sign_positive() {
-            Some(1)
-        } else {
-            Some(-1)
-        }
-    } else if abs_diff_eq!(value, 0.0) {
+    if value.abs() <= f32::EPSILON {
         Some(0)
+    } else if value.abs() - 1.0 <= f32::EPSILON {
+        let one: f32 = 1.0;
+        Some(one.copysign(value) as i32)
```

Comment:
```suggestion
        Some(1.0f32.copysign(value) as i32)
```

or maybe

```suggestion
        Some(1 * value.signum() as i32)
```

When I made this suggestion, I don't think I realized that the return value was `i32`. Whichever of these makes the most sense to you should be good!